api/views/__init__.py

```diff
@@ -195,6 +196,46 @@ def runtime_data(request):
     )
 
 
+class FlagFilter(filters.FilterSet):
+    class Meta:
+        model = get_waffle_flag_model()
+        fields = [
+            "name",
+            "everyone",
+            # "users",
+            # read-only
+            "id",
+        ]
+
+
+class FlagViewSet(SaveToRequestUser, viewsets.ModelViewSet):
+    serializer_class = FlagSerializer
+    permission_classes = [permissions.IsAuthenticated]
+    filterset_class = FlagFilter
+
+    def get_queryset(self):
+        if not flag_is_active(
+            self.request, "manage_flags"
+        ) or not self.request.user.email.endswith("@mozilla.com"):
```

Comment:
Ah sorry, the negative disjunction here is confusing, it should be clearer in eaa4dd4f. But not everyone at Mozilla would have the ability to modify flags; only those who have the `manage_flags` permission. So the idea would be to give that flag to a limited set of accounts, _and_ require that those accounts have `@mozilla.com` email addresses.

And since the API doesn't allow people to modify the `manage_flags` flag, only SRE would be able to give people that ability (so SREs would be "flag owners", and they can make people "flag managers"). There would be no "flag users" yet.

I'd be happy to have this managed by Django groups and permissions, which indeed sounds like the more appropriate tool for the job. And I think that might actually be what @say-yawn is looking at for the "real" solution, which would use the admin UI?